test/TestTextSerialisation.hs

```diff
@@ -6,11 +6,17 @@ import Test.Framework.Providers.HUnit (testCase)
 import Blaze.ByteString.Builder (toByteString)
 
 import Clapi.Types (Time(..), ClapiValue(..))
-import Clapi.TextSerialisation (encode)
+import Clapi.TextSerialisation (encode, decode)
 
 tests = [
-   testCase "basic text encode" testBasicEncode
+   testCase "basic text encode" testBasicEncode,
+   testCase "basic text decode" testBasicDecode
    ]
 
-testBasicEncode = assertEqual "encode result" "Bis\n0:0 T 3 \"marimba\"" $
-    toByteString . encode $ Map.fromList [(Time 0 0, [ClBool True, ClInt32 3, ClString "marimba"])]
+timeseries = Map.fromList [(Time 0 0, [ClBool True, ClInt32 3, ClString "marimba"])]
+encoded _ = "Bis\n0:0 T 3 \"marimba\""
```

Comment:
A reason that this is not polymorphic if you don't parametrise it can be found [here](http://w3cgeek.com/why-are-polymorphic-values-not-inferred-in-haskell.html). Seeing as all top-level definitions should, by good practice, have a type definition anyway, you could so something like:
```haskell
import Data.String (IsString)
...
encoded :: (IsString a) => a
encoded = "foo"
```
which at least rids you of the undefined.